pkg/systemd/quadlet/quadlet.go

```diff
@@ -416,14 +428,19 @@ func ConvertContainer(container *parser.UnitFile, names map[string]string, isUse
 
 	// If conmon exited uncleanly it may not have removed the container, so
 	// force it, -i makes it ignore non-existing files.
-	service.Add(ServiceGroup, "ExecStop", podmanBinary()+" rm -v -f -i --cidfile=%t/%N.cid")
+	serviceStopCmd := createBasePodmanCommand(container, ContainerGroup)
+	serviceStopCmd.add("rm", "-v", "-f", "-i", "--cidfile=%t/%N.cid")
+	service.AddCmdline(ServiceGroup, "ExecStop", serviceStopCmd.Args)
 	// The ExecStopPost is needed when the main PID (i.e., conmon) gets killed.
 	// In that case, ExecStop is not executed but *Post only.  If both are
 	// fired in sequence, *Post will exit when detecting that the --cidfile
 	// has already been removed by the previous `rm`..
-	service.Add(ServiceGroup, "ExecStopPost", "-"+podmanBinary()+" rm -v -f -i --cidfile=%t/%N.cid")
+	serviceStopCmd.Args = append([]string{"-"}, serviceStopCmd.Args...)
```

Comment:
This change does not look right, it will result in `- binary` not the old (correct) `-binary`, notice the extra space.

The fact that none of the test catches that worries me, testing locally I see
```
Empty path in command line, ignoring: '- /usr/bin/podman rm -v -f -i --cidfile=%t/%N.cid'
``` 
as warning in the journal so it is not accepted by systemd.

Reply:
Thanks for finding this.
The issue was in the regex being checked as it was not forcing non white space after the `-` sign. I've fixed it and the code.